app/models/user.rb

```diff
@@ -111,4 +134,28 @@ def initialize_password
 
     self.password ||= SecureRandom.base64(8) + required_chars.join
   end
+
+  def selectable_first_year(target_date)
+    entry_date.strftime('%m%d') <= target_date.strftime('%m%d') ? entry_date.year : entry_date.year + 1
+  end
+
+  def selectable_last_year(target_date)
+    target_date.strftime('%m%d') <= Date.today.strftime('%m%d') ? Date.today.year : Date.today.year - 1
+  end
+
+  def selectable_first_month(target_date)
+    target_date.year == entry_date.year ? entry_date.month : 1
+  end
+
+  def selectable_last_month(target_date)
```

Comment:
__[reek]__
* [UtilityFunction] doesn't depend on instance state (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Utility-Function.md)

<!-- Comment by [SideCI](https://sideci.com) -->